### PR TITLE
Pipe

### PR DESCRIPTION
**Pull request by joredudiaz**
## Summary

- Filter books by a custom filter pipe

## Jira Story

https://trello.com/c/sj7CpBr1

## Screenshot

pipe

### CODE REVIEW

File: wolox-books/src/app/pipes/filter-book.pipe.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { Book } from '../screens/book/book';
+
+@Pipe({
+  name: 'filterBook'
+})
+export class FilterBookPipe implements PipeTransform {
+
+  transform(books: Book[], search: string): any {
+    if(!!search && search.trim() != "") {
```
Comment (FrankIglesias) [inline wolox-books/src/app/pipes/filter-book.pipe.ts]:
```suggestion
    if(!!search && search.trim() !== "") {
```

Comment (FrankIglesias) [inline wolox-books/src/app/pipes/filter-book.pipe.ts]: you can remove this if

File: wolox-books/src/app/store/book.reducer.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { Book } from '../screens/book/book';
 import * as BookActions from './book.action';
 
-const initialState = [];
+const initialState = [ ];
```
Comment (FrankIglesias) [inline wolox-books/src/app/store/book.reducer.ts]: remove empty space

**Review by FrankIglesias — CHANGES_REQUESTED** [review]

File: wolox-books/src/app/pipes/filter-book.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+      return books.filter(book => book.title.toLowerCase() == search.toLocaleLowerCase());
```
Comment (betoobernal) [inline wolox-books/src/app/pipes/filter-book.pipe.ts]:
```suggestion
      return books.filter(book => book.title.toLowerCase() === search.toLocaleLowerCase());
```

File: wolox-books/src/app/screens/book/components/list/list.component.html [hunk 4]
```diff
@@ -1,8 +1,8 @@
 <div class="search">
-  <input type="text" class="input-search" placeholder="Busc&aacute; por t&iacute;tulo del libro...">
+  <input type="text" class="input-search" placeholder="Busc&aacute; por t&iacute;tulo del libro..." [(ngModel)]="search">
 </div>
 
-<div *ngFor="let book of books | async" class="book-container">
+<div *ngFor="let book of books | async | filterBook:search" class="book-container">
   <div class="book-card" *ngFor="let number of [0,1,2,3,4,5]">
     <img src="../../../../../assets/book-cover.png" alt="" class="book-cover">
```
Comment (betoobernal) [inline wolox-books/src/app/screens/book/components/list/list.component.html]: close tag

File: wolox-books/src/app/pipes/filter-book.pipe.ts (hunk 1 above)
Comment (joredudiaz) [reply wolox-books/src/app/pipes/filter-book.pipe.ts]: I can't, if I remove the `if`, the filter search by an empty string, and wont show anything

File: wolox-books/src/app/screens/book/components/list/list.component.html (hunk 4 above)
Comment (joredudiaz) [reply wolox-books/src/app/screens/book/components/list/list.component.html]: raise an angular error as the latest PRs

Comment (betoobernal) [reply wolox-books/src/app/screens/book/components/list/list.component.html]: I don't remind that error 😅

**Review by FrankIglesias — APPROVED** [review]

**Review by betoobernal — APPROVED** [review]

**Pull request merged**